subsystems/viamserver/viamserver.go

```diff
@@ -102,6 +105,15 @@ func (s *viamServer) Start(ctx context.Context) error {
 	s.cmd.Stdout = stdio
 	s.cmd.Stderr = stderr
 
+	if len(s.extraEnvVars) > 0 {
+		// if s.cmd.Env is not explicitly specified (nil), viam-server would inherit all env vars in Agent's environment
+		s.cmd.Env = s.cmd.Environ()
+		for k, v := range s.extraEnvVars {
+			s.cmd.Env = append(s.cmd.Env, k+"="+v)
+		}
+		s.logger.Debugf("Starting viam-server with environment variables %v", s.cmd.Env)
```

Comment:
let's add an Info log for `s.extraEnvVars` and we can keep the debug log for the full env